x-pack/apm-server/sampling/processor_test.go

```diff
@@ -905,3 +905,11 @@ func waitFileModified(tb testing.TB, filename string, after time.Time) ([]byte,
 		}
 	}
 }
+
+func cleanupProcessor(t *testing.T, processor *sampling.Processor) {
+	t.Cleanup(func() {
+		ctx, cancel := context.WithTimeout(context.Background(), time.Millisecond)
+		defer cancel()
+		processor.Stop(ctx)
```

Comment:
Is there even a point in waiting 1ms? Should we use WithCancel and immediately cancel the context?

Reply:
I did that initially but got a panic, so I decided to use 1ms which shouldn't really affect the test runtime anyway. I haven't looked into what exactly triggers this panic, though.

```
panic: Transaction has already been discarded

goroutine 102526 [running]:
github.com/dgraph-io/badger/v2.(*Txn).NewIterator(0x1a8317f4cb1?, {0x1, 0x64, 0x0, 0x0, {0x140000438c0, 0x25, 0x30}, 0x0, 0x0})
	/Users/marclop/go/pkg/mod/github.com/dgraph-io/badger/v2@v2.2007.3-0.20201012072640-f5a7e0a1c83b/iterator.go:460 +0x4b4
github.com/elastic/apm-server/x-pack/apm-server/sampling/eventstorage.(*ReadWriter).ReadTraceEvents(0x14000f17440, {0x14000043890?, 0x24}, 0x14000650048)
	/Users/marclop/repos/elastic/apm-server/x-pack/apm-server/sampling/eventstorage/storage.go:220 +0x170
github.com/elastic/apm-server/x-pack/apm-server/sampling/eventstorage.(*lockedReadWriter).ReadTraceEvents(0x14000ce1530?, {0x14000043890?, 0x18?}, 0x105cdf500?)
	/Users/marclop/repos/elastic/apm-server/x-pack/apm-server/sampling/eventstorage/sharded.go:110 +0xd4
github.com/elastic/apm-server/x-pack/apm-server/sampling/eventstorage.(*ShardedReadWriter).ReadTraceEvents(0x14001080b58?, {0x14000043890, 0x24}, 0x28?)
	/Users/marclop/repos/elastic/apm-server/x-pack/apm-server/sampling/eventstorage/sharded.go:56 +0x3c
github.com/elastic/apm-server/x-pack/apm-server/sampling.(*wrappedRW).ReadTraceEvents(...)
	/Users/marclop/repos/elastic/apm-server/x-pack/apm-server/sampling/processor.go:586
github.com/elastic/apm-server/x-pack/apm-server/sampling.(*Processor).Run.func7()
	/Users/marclop/repos/elastic/apm-server/x-pack/apm-server/sampling/processor.go:465 +0x1e8
golang.org/x/sync/errgroup.(*Group).Go.func1()
	/Users/marclop/go/pkg/mod/golang.org/x/sync@v0.0.0-20220819030929-7fc1605a5dde/errgroup/errgroup.go:75 +0x60
created by golang.org/x/sync/errgroup.(*Group).Go
	/Users/marclop/go/pkg/mod/golang.org/x/sync@v0.0.0-20220819030929-7fc1605a5dde/errgroup/errgroup.go:72 +0xa8
```